helm/aladdin-demo/templates/aladdin-demo-elasticsearch.statefulset.yaml

```diff
@@ -0,0 +1,78 @@
+{{ if .Values.elasticsearch.create }}
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: db
+  labels:
+    project: {{ .Chart.Name }}
+    name: {{ .Chart.Name }}-elasticsearch
+    app: {{ .Chart.Name }}-elasticsearch
+    githash: {{.Values.deploy.imageTag}}
+spec:
+  selector:
+    matchLabels:
+      name: {{ .Chart.Name }}-elasticsearch
+  # The name of the service that governs this StatefulSet, must exist before the StatefulSet    
+  serviceName: {{ .Chart.Name }}-elasticsearch
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: {{ .Chart.Name }}-elasticsearch # has to match .spec.selector.matchLabels
+    spec:
+      initContainers:
+      # Increase the vm.max_map_count to satisfy the bootstrap checks of elasticsearch
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      # Change the owner of the mounted persistent volume let elasticsearch access it for data and logs
+      # Ideally this can be done with spec.securityContext, which is commented out below, but it 
+      # does not appear to work in minikube
+      - name: volume-mount-hack
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sh", "-c", "chown -R {{ .Values.elasticsearch.id }}:{{ .Values.elasticsearch.id }} /stateful"]
+        volumeMounts:
+        - name: storage
+          mountPath: /stateful
```

Comment:
to keep things consistent, we can keep mount path same as main pod i.e. /usr/share/elasticsearch/stateful
It makes it easier to read.